sys/net/gnrc/transport_layer/udp/gnrc_udp.c

```diff
@@ -130,9 +130,9 @@ static void _receive(gnrc_pktsnip_t *pkt)
     port = (uint32_t)byteorder_ntohs(hdr->dst_port);
 
     /* send payload to receivers */
-    if (!gnrc_netapi_dispatch_receive(GNRC_NETTYPE_UDP, port, pkt)) {
+    if (!gnrc_netapi_dispatch_receive(GNRC_NETTYPE_UDP, port, udp)) {
```

Comment:
Well in current master it is the payload (pktsnip of type undef). I think this this is what is actually right. And I believe the solution that I presented here is wrong :/